core/src/main/java/org/elasticsearch/common/joda/DateMathParser.java

```diff
@@ -190,6 +190,11 @@ private long parseMath(String mathString, long time, boolean roundUp, DateTimeZo
     private long parseDateTime(String value, DateTimeZone timeZone, boolean roundUpIfNoTime) {
         DateTimeFormatter parser = dateTimeFormatter.parser();
         if (timeZone != null) {
+            String format = dateTimeFormatter.format();
+            // Non UTC time zones are not compatible with epoch formats
+            if (timeZone != DateTimeZone.UTC && format.equals("epoch_second") || format.equals("epoch_millis")) {
```

Comment:
Nice catch I have added some test cases explicitly passing in UTC to make sure this is caught in the future.